Modules/cPickle.c

```diff
@@ -607,6 +609,10 @@ readline_file(Unpicklerobject *self, char **s)
         self->buf = newbuf;
         self->buf_size = bigger;
     }
+done:
```

Comment:
I find it a little weird that this section is down here when the only way you can get to it is to jump out of a twice nested loop above.

Reply:
In my mind the end of the function is the natural place to put this, because it makes clear that the outer loop doesn't end without this bit getting executed (barring error-returns).